### PR TITLE
Add HierarchicalNSW::indexFileSize() function for precise memory footprint control

### PR DESCRIPTION
**Pull request by drons**

### CODE REVIEW

**Comment by yurymalkov** [comment]
Hi @drons,
Thank for the PR! We will check it and merge it after we prepare the release (should happen soon).

File: python_bindings/bindings.cpp [hunk 1]
```diff
@@ -218,6 +218,9 @@ class Index {
         this->num_threads_default = num_threads;
     }
 
+    void indexFileSize() const {
```
Comment (dyashuni) [inline python_bindings/bindings.cpp:221]: we need to return an index size here
```cpp
size_t indexFileSize() const {
    return appr_alg->indexFileSize();
}
```

**Review by dyashuni — CHANGES_REQUESTED** [review]
Thank you, please add a small change

File: python_bindings/bindings.cpp (hunk 1 above)
Comment (drons) [reply python_bindings/bindings.cpp]: :man_facepalming:

**Review by yurymalkov — APPROVED** [review]

**Comment by yurymalkov** [comment]
Thank you for the PR!

**Comment by drons** [comment]
Thanks!

**Pull request merged**